In [1]:
import math

In [18]:
points = [(3,2),(4,1),(12,8),(2,6)]

In [19]:
distances = []
def euclideanDistance():#Argüman almadan hesaplama
    distance = 0
    for i in range(len(points)-1):
        x1, y1 = points[i]
        for j in range(i+1,len(points)):
            x2,y2 = points[j]
            distance = math.sqrt(pow((x2-x1),2)+pow((y2-y1),2))
            distances.append(distance)      
    return distances

In [20]:
distances_2 = euclideanDistance()
distances_2

[1.4142135623730951,
 10.816653826391969,
 4.123105625617661,
 10.63014581273465,
 5.385164807134504,
 10.198039027185569]

In [21]:
distances_2.sort()
distances_2

[1.4142135623730951,
 4.123105625617661,
 5.385164807134504,
 10.198039027185569,
 10.63014581273465,
 10.816653826391969]

nokta_1: (3, 2)
------
nokta_2:(4, 1)
 (x1,y1):(3, 2) ve (x2,y2):(4, 1) noktalarının mesafesi:1.4142135623730951
------
nokta_2:(12, 8)
 (x1,y1):(3, 2) ve (x2,y2):(12, 8) noktalarının mesafesi:10.816653826391969
------
nokta_2:(2, 6)
 (x1,y1):(3, 2) ve (x2,y2):(2, 6) noktalarının mesafesi:4.123105625617661
nokta_1: (4, 1)
------
nokta_2:(12, 8)
 (x1,y1):(4, 1) ve (x2,y2):(12, 8) noktalarının mesafesi:10.63014581273465
------
nokta_2:(2, 6)
 (x1,y1):(4, 1) ve (x2,y2):(2, 6) noktalarının mesafesi:5.385164807134504
nokta_1: (12, 8)
------
nokta_2:(2, 6)
 (x1,y1):(12, 8) ve (x2,y2):(2, 6) noktalarının mesafesi:10.198039027185569